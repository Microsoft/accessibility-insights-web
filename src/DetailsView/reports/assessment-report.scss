// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';
@import './components/outcome.scss';
@import './components/assessment-report-header.scss';
@import './components/assessment-report-summary.scss';
@import './components/assessment-summary-details.scss';
@import './components/assessment-scan-details.scss';
@import './components/assessment-report-body-header.scss';

* {
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    -webkit-print-color-adjust: exact;
    /* Chrome, Safari */
    color-adjust: exact;
    /*Firefox*/
}

body {
    max-width: 768px;
    min-width: 480px;
    margin: auto;
}

header {
    margin-bottom: 24px;
}

.details-section {
    border-bottom: $neutral-alpha-20 solid 1px;
}

.details-section-header {
    margin: 0px 0px 32px 0px;
    display: flex;
    align-items: center;
    font-size: 21px;
    font-family: Segoe UI Semibold;
    line-height: 32px;
}

.assessment-details {
    margin: 0px 0px 32px 0px;
}

.assessment-header {
    margin: 0px 0px 32px 0px;
    flex-wrap: wrap;
    display: flex;
    align-items: center;
    font-size: 17px;
    font-family: Segoe UI Semibold;
}

.step-details {
    margin: 0px 0px 24px 48px;

    &:last-child {
        margin-bottom: 0px;
    }
}

.step-header {
    padding-bottom: 12px;
}

.step-header-name {
    font-size: 15px;
    font-family: Segoe UI;
    display: inline;
    margin-block-start: 0;
    margin-block-end: 0;
    font-weight: normal;
}

.step-header-description {
    font-size: 15px;
    font-family: Segoe UI;
    padding-left: 8px;
    padding-right: 8px;
    display: inline;
}

.step-header-message {
    font-size: 15px;
    font-family: Segoe UI;
    padding-left: 8px;
    padding-right: 8px;
    display: inline;
}

.instance-details {
    padding-left: 12px;
    padding-top: 12px;
    width: 100%;

    &:not(:last-child) {
        padding-bottom: 14px;
        border-bottom: $neutral-8 solid 1px;
    }
}

.instance-pair-details > td {
    padding-bottom: 8px;
    line-height: 16px;
}

.instance-value {
    font-size: 12px;
    font-family: Segoe UI;
    vertical-align: top;
    word-break: break-all;
}

.instance-key {
    font-size: 12px;
    font-family: Segoe UI Semibold;
    vertical-align: top;
    text-align: left;
    width: 140px;
}

.instance-subsection-header {
    @extend .instance-key;
    text-align: left;
}

.instance-key-indented {
    @extend .instance-key;
    padding-left: 30px;
}

.test-guidance-links-group {
    display: inline;
    color: $primary-text;
    font-size: 14px;
    font-family: Segoe UI;
    padding-right: 8px;
    padding-left: 8px;
}

.report-footer {
    font-size: 14px;
    font-family: Segoe UI;
    color: $secondary-text;
    padding-top: 24px;
    padding-bottom: 36px;
    padding-left: 12px;
    line-height: 20px;
}

.report-footer-link {
    font-size: 14px;
    font-family: Segoe UI;
    text-decoration: none;
}

.ms-Link,
.link {
    color: $communication-shade-10;
    &:hover {
        color: $communication-shade-30;
    }
}

@media print {
    .ms-Link,
    .link {
        text-decoration: none;
        color: $primary-text;
    }
}
