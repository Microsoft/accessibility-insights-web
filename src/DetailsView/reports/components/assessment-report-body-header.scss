// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../../common/styles/colors.scss';

.assessment-report-body-header {
    margin: 0px 0px 21px 0px;

    h1 {
        font-size: 28px;
        margin: 0px 0px 24px 0px;
        font-family: Segoe UI Semibold;
        line-height: 40px;
        letter-spacing: -1;
    }

    p {
        font-size: 14px;
        color: $secondary-text;
        font-family: Segoe UI;
        margin: 0;
        line-height: 20px;
    }
}
