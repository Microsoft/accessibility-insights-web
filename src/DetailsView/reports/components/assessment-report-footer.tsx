// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { title } from '../../../content/strings/application';

export interface AssessmentReportFooterProps {
    extensionVersion: string;
    axeVersion: string;
    chromeVersion: string;
}

export class AssessmentReportFooter extends React.Component<AssessmentReportFooterProps> {
    public render(): JSX.Element {
        return (
            <footer className="report-footer">
                This assessment report was generated using {title} {this.props.extensionVersion} (Axe {this.props.axeVersion}), a tool that
                helps debug and find accessibility issues earlier on {this.props.chromeVersion}. Get more information & download this tool
                at{' '}
                <a href="http://aka.ms/AccessibilityInsights" className="link report-footer-link" target="_blank">
                    http://aka.ms/AccessibilityInsights
                </a>
            </footer>
        );
    }
}
