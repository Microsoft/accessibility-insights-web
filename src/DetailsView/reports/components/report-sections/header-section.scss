// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../../../common/styles/colors.scss';
@import '../../../../common/styles/fonts.scss';

.report-header-bar svg {
    margin-left: 8px;
}

.report-header-command-bar {
    height: 40px;
    width: 100%;
    display: flex;
    justify-content: end;
    min-height: fit-content;
    align-items: center;
    background-color: $neutral-0;
    box-shadow: 0px 1px 0px rgba(0, 0, 0, 0.08);

    .target-page {
        margin: 0px 0px 0px 16px;
        display: inherit;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
        color: $neutral-60;
        font-family: $fontFamily;
        font-size: 14px;
        line-height: 20px;
    }
}
