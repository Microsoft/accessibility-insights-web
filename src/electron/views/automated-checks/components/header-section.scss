// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../../../common/styles/colors.scss';

.h1 {
    font-size: 21px;
    line-height: 25px;
    letter-spacing: -0.02em;
}

.text {
    font-size: 14px;
    line-height: 16px;
    color: $secondary-text;
}
