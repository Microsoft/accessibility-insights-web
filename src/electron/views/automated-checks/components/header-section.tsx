// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';

export const HeaderSection = NamedFC('HeaderSection', () => {
    return (
        <>
            <h1>Automated checks</h1>
            <span>
                Automated checks can detect some common accessibility problems such as missing or invalid properties. But most accessibility
                problems can only be discovered through manual testing.
            </span>
        </>
    );
});
