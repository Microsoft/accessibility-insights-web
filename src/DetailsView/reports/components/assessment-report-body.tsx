// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { ManualTestStatus } from '../../../common/types/manual-test-status';
import { IAssessmentDetailsReportModel, IReportModel } from '../assessment-report-model';
import { AssessmentReportAssessmentList } from './assessment-report-assessment-list';
import { AssessmentReportBodyHeader } from './assessment-report-body-header';
import { AssessmentReportSummary } from './assessment-report-summary';
import { AssessmentScanDetails } from './assessment-scan-details';
import { OutcomeChip } from './outcome-chip';
import { allOutcomeTypes } from './outcome-type';

export interface IAssessmentReportBodyProps {
    data: IReportModel;
    description: string;
}

export class AssessmentReportBody extends React.Component<IAssessmentReportBodyProps> {
    public render(): JSX.Element {
        return (
            <div className="assessment-report-body" role="main">
                <AssessmentReportBodyHeader />
                <AssessmentReportSummary summary={this.props.data.summary} />
                <AssessmentScanDetails details={this.props.data.scanDetails} description={this.props.description} />

                {this.renderDetailsSection(
                    this.props.data.failedDetailsData,
                    'Failed tests',
                    this.props.data.summary.byRequirement.fail,
                    ManualTestStatus.FAIL,
                )}

                {this.renderDetailsSection(
                    this.props.data.incompleteDetailsData,
                    'Incomplete tests',
                    this.props.data.summary.byRequirement.incomplete,
                    ManualTestStatus.UNKNOWN,
                )}

                {this.renderDetailsSection(
                    this.props.data.passedDetailsData,
                    'Passed tests',
                    this.props.data.summary.byRequirement.pass,
                    ManualTestStatus.PASS,
                )}
            </div>
        );
    }

    private renderDetailsSection(
        detailsData: IAssessmentDetailsReportModel[],
        title: string,
        count: number,
        status: ManualTestStatus,
    ): JSX.Element {
        return (
            <div className="details-section">
                {this.renderDetailsSectionHeader(title, count, status)}
                <AssessmentReportAssessmentList status={status} assessments={detailsData} />
            </div>
        );
    }
    private renderDetailsSectionHeader(title: string, count: number, status: ManualTestStatus): JSX.Element {
        return (
            <div className="details-section-header">
                <h2 className="details-section-header-title">{title}</h2>

                <OutcomeChip count={count} outcomeType={allOutcomeTypes[status]} />
            </div>
        );
    }
}
