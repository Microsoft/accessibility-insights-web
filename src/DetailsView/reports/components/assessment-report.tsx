// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { IReportModel } from '../assessment-report-model';
import { AssessmentReportBody, AssessmentReportBodyDeps } from './assessment-report-body';
import { AssessmentReportFooter } from './assessment-report-footer';
import { AssessmentReportHeader } from './assessment-report-header';

export type AssessmentReportDeps = AssessmentReportBodyDeps;

export interface AssessmentReportProps {
    deps: AssessmentReportDeps;
    data: IReportModel;
    description: string;
    extensionVersion: string;
    axeVersion: string;
    chromeVersion: string;
}

export class AssessmentReport extends React.Component<AssessmentReportProps> {
    public render(): JSX.Element {
        return (
            <React.Fragment>
                <AssessmentReportHeader
                    targetPageTitle={this.props.data.scanDetails.targetPage}
                    targetPageUrl={this.props.data.scanDetails.url}
                />
                <AssessmentReportBody
                    deps={this.props.deps}
                    data={this.props.data}
                    description={this.props.description}
                />

                <AssessmentReportFooter
                    extensionVersion={this.props.extensionVersion}
                    axeVersion={this.props.axeVersion}
                    chromeVersion={this.props.chromeVersion}
                />
            </React.Fragment>
        );
    }
}
