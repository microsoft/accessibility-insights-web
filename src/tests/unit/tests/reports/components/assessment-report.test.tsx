// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import * as React from 'react';
import { AssessmentReport, AssessmentReportDeps } from 'reports/components/assessment-report';
import { AssessmentReportBodyHeader } from 'reports/components/assessment-report-body-header';
import { AssessmentReportBody } from '../../../../../reports/components/assessment-report-body';
import { AssessmentReportFooter } from '../../../../../reports/components/assessment-report-footer';
import { HeaderSection } from '../../../../../reports/components/report-sections/header-section';
import {
    expectMockedComponentPropsToMatchSnapshots,
    mockReactComponents,
} from '../../../mock-helpers/mock-module-helpers';
import { AssessmentReportBuilderTestHelper } from '../../DetailsView/assessment-report-builder-test-helper';
jest.mock('../../../../../reports/components/report-sections/header-section');
jest.mock('../../../../../reports/components/assessment-report-body');
jest.mock('../../../../../reports/components/assessment-report-footer');

describe('AssessmentReport', () => {
    mockReactComponents([HeaderSection, AssessmentReportBody, AssessmentReportFooter]);
    test('render', () => {
        const deps: AssessmentReportDeps = {
            outcomeTypeSemanticsFromTestStatus: {
                stub: 'outcomeTypeSemanticsFromTestStatus',
            } as any,
        } as AssessmentReportDeps;

        const data = AssessmentReportBuilderTestHelper.getAssessmentReportModel();

        const bodyHeader = <AssessmentReportBodyHeader />;

        const renderResult = render(
            <AssessmentReport
                deps={deps}
                data={data}
                bodyHeader={bodyHeader}
                description="test string"
                extensionVersion="ProductVersion"
                axeVersion="axeVersion"
                chromeVersion="chromeVersion"
            />,
        );
        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([HeaderSection, AssessmentReportBody]);
    });
});
