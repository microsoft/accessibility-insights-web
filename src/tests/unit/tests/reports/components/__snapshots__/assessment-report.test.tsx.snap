// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentReport render 1`] = `
<React.Fragment>
  <HeaderSection
    headerText="Accessibility Insights for Web"
    targetAppInfo={
      {
        "name": "title",
        "url": "url",
      }
    }
  />
  <AssessmentReportBody
    data={
      {
        "failedDetailsData": [
          {
            "displayName": "Assessment1",
            "key": "assessment1",
            "steps": [
              {
                "defaultMessageComponent": null,
                "header": {
                  "description": null,
                  "displayName": "Step2A",
                  "guidanceLinks": [],
                  "requirementType": "assisted",
                },
                "instances": [
                  {
                    "props": [
                      {
                        "key": "Path",
                        "value": "target1",
                      },
                      {
                        "key": "Snippet",
                        "value": "someHtml",
                      },
                    ],
                  },
                ],
                "key": "step2a",
                "showPassingInstances": false,
              },
              {
                "defaultMessageComponent": null,
                "header": {
                  "description": null,
                  "displayName": "Step4A",
                  "guidanceLinks": [],
                  "requirementType": "manual",
                },
                "instances": [
                  {
                    "props": [
                      {
                        "key": "Comment",
                        "value": "comment",
                      },
                    ],
                  },
                ],
                "key": "step4a",
                "showPassingInstances": false,
              },
            ],
          },
          {
            "displayName": "Assessment2",
            "key": "assessment2",
            "steps": [
              {
                "defaultMessageComponent": null,
                "header": {
                  "description": null,
                  "displayName": "Step2B",
                  "guidanceLinks": [],
                  "requirementType": "manual",
                },
                "instances": [],
                "key": "step2b",
                "showPassingInstances": true,
              },
            ],
          },
        ],
        "incompleteDetailsData": [
          {
            "displayName": "Assessment1",
            "key": "assessment1",
            "steps": [
              {
                "defaultMessageComponent": null,
                "header": {
                  "description": null,
                  "displayName": "Step3A",
                  "guidanceLinks": [],
                  "requirementType": "assisted",
                },
                "instances": [
                  {
                    "props": [
                      {
                        "key": "Path",
                        "value": "target2",
                      },
                      {
                        "key": "Snippet",
                        "value": "someHtml",
                      },
                    ],
                  },
                ],
                "key": "step3a",
                "showPassingInstances": false,
              },
            ],
          },
          {
            "displayName": "Assessment2",
            "key": "assessment2",
            "steps": [
              {
                "defaultMessageComponent": null,
                "header": {
                  "description": null,
                  "displayName": "Step1B",
                  "guidanceLinks": [],
                  "requirementType": "assisted",
                },
                "instances": [],
                "key": "step1b",
                "showPassingInstances": true,
              },
            ],
          },
        ],
        "passedDetailsData": [
          {
            "displayName": "Assessment1",
            "key": "assessment1",
            "steps": [
              {
                "defaultMessageComponent": null,
                "header": {
                  "description": null,
                  "displayName": "Step1A",
                  "guidanceLinks": [
                    {
                      "href": "link1A",
                      "text": "link1A",
                    },
                  ],
                  "requirementType": "assisted",
                },
                "instances": [
                  {
                    "props": [
                      {
                        "key": "Path",
                        "value": "target1",
                      },
                      {
                        "key": "Snippet",
                        "value": "someHtml",
                      },
                    ],
                  },
                ],
                "key": "step1a",
                "showPassingInstances": false,
              },
            ],
          },
        ],
        "scanDetails": {
          "reportDate": 2000-01-01T00:00:00.000Z,
          "targetPage": "title",
          "url": "url",
        },
        "summary": {
          "byPercentage": {
            "fail": 50,
            "incomplete": 37,
            "pass": 13,
          },
          "byRequirement": {
            "fail": 3,
            "incomplete": 2,
            "pass": 1,
          },
          "reportSummaryDetailsData": [
            {
              "displayName": "Assessment1",
              "fail": 2,
              "incomplete": 1,
              "pass": 1,
            },
            {
              "displayName": "Assessment2",
              "fail": 1,
              "incomplete": 1,
              "pass": 0,
            },
          ],
        },
      }
    }
    deps={
      {
        "outcomeTypeSemanticsFromTestStatus": {
          "stub": "outcomeTypeSemanticsFromTestStatus",
        },
      }
    }
    description="test string"
  />
  <AssessmentReportFooter
    axeVersion="axeVersion"
    chromeVersion="chromeVersion"
    extensionVersion="ProductVersion"
  />
</React.Fragment>
`;
